.github/workflows/ci.yaml

```diff
@@ -82,10 +82,47 @@ jobs:
       run: echo "##[set-output name=imagetag;]$(echo ${GITHUB_REF##*/})"
       id: extract_tag_name
 
-    - name: Build funcX-endpoint Image
+    - name: Build funcX-endpoint Image for python 3.6
       uses: elgohr/Publish-Docker-Github-Action@master
+      env:
+        PYTHON_VERSION: 3.6
       with:
         dockerfile: Dockerfile-endpoint
-        name: funcx/kube-endpoint:${{ steps.extract_tag_name.outputs.imagetag }}
+        name: funcx/kube-endpoint:${{ steps.extract_tag_name.outputs.imagetag }}-3.6
         username: ${{ secrets.DOCKER_USERNAME }}
         password: ${{ secrets.DOCKER_PASSWORD }}
+        buildargs: PYTHON_VERSION
+
+    - name: Build funcX-endpoint Image for python 3.7
```

Comment:
You can cut down some of the extra steps by [using GH's matrix keyword.](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstrategymatrix) The end result would be a workflow that looks like:
```
  publish:
    needs: test
    runs-on: ubuntu-latest
    strategy:
        matrix:
            python: [3.6, 3.7, 3.8, 3.9]
    steps:
        <parameterized on python version>
```

Reply:
Genius!